spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/VaultEnvironmentRepository.java

```diff
@@ -204,23 +230,37 @@ private void addProfiles(List<String> contexts, String baseContext,
 		}
 	}
 
-	private String read(String key) {
-		HttpHeaders headers = new HttpHeaders();
-		headers.add(VAULT_TOKEN, getToken());
-		if (StringUtils.hasText(this.namespace)) {
-			headers.add(VAULT_NAMESPACE, this.namespace);
-		}
-
-		return this.accessStrategy.getData(headers, this.backend, key);
+	private String read(String key, HttpHeaders vaultHeaders) {
+		return this.accessStrategy.getData(vaultHeaders, this.backend, key);
 	}
 
-	private String getToken() {
-		String token = tokenProvider.getToken();
-		if (!StringUtils.hasLength(token)) {
-			throw new IllegalArgumentException(
-					"A Vault token must be supplied by a token provider");
+	private String getToken(HttpHeaders vaultHeaders) {
+		String vaultToken = tokenProvider.getToken();
+		if (!StringUtils.hasLength(vaultToken)) {
+			HttpServletRequest servletRequest = this.request.getIfAvailable();
```

Comment:
All of the token retrieval logic was previously moved from `VaultEnvironmentRepository` to the `ConfigTokenProvider` implementations to better support the embedded config server use case (where there is no `HttpServletRequest` to get auth info from). This change re-introduces the `HttpServletRequest` requirement into `VaultEnvironmentRepository`, which is a step backwards IMO.

Reply:
Good catch. There will need to be an analogous interface for this.